### PR TITLE
Node size based on nucleotides

### PR DESCRIPTION
**Pull request by niekvanderlaan**

### CODE REVIEW

**Comment by ric-jon** [comment]
Needs more checkstyle

**Comment by ghost** [comment]
Merge conflict needs to be fixed before merging

File: src/main/java/application/controllers/LegendFactory.java [hunk 1]
```diff
@@ -27,10 +27,10 @@ public LegendFactory() {
      */
     public HBox createLegend() {
         final VBox col1 = new VBox();
-        col1.getChildren().add(new RectangleCell(0));
-        col1.getChildren().add(new BubbleCell(0, "N"));
-        col1.getChildren().add(new IndelCell(0, "N"));
-        col1.getChildren().add(new CollectionCell(0, "N"));
+        col1.getChildren().add(new RectangleCell(0, 1));
+        col1.getChildren().add(new BubbleCell(0, 1, "N"));
+        col1.getChildren().add(new IndelCell(0,1, "N"));
```
Comment (ghost) [inline src/main/java/application/controllers/LegendFactory.java]: Will cause checkstyle violation

File: src/main/java/application/fxobjects/cell/graph/BubbleCell.java [hunk 2]
```diff
@@ -36,16 +36,14 @@ public BubbleCell(int id, String collapseLevel) {
      * @param pane A given stack pane.
      * @param text A given text element.
      */
-    public BubbleCell(int id, StackPane pane, Text text) {
+    public BubbleCell(int id, int nucleotides, StackPane pane, Text text) {
         super(id);
-
-        shape = new Circle(10);
+        shape = new Circle(Math.min(10.0 + ((double)nucleotides) / 80000, 100));
```
Comment (ghost) [inline src/main/java/application/fxobjects/cell/graph/BubbleCell.java]: Probable useless parenthesis

File: src/main/java/application/fxobjects/cell/graph/CollectionCell.java [hunk 3]
```diff
@@ -25,8 +25,8 @@
      * @param id            The ID of a cell.
      * @param collapseLevel The collapse level of a cell.
      */
-    public CollectionCell(int id, String collapseLevel) {
-        this(id, new StackPane(), new Text(collapseLevel));
+    public CollectionCell(int id, int nucleotides, String collapseLevel) {
```
Comment (ghost) [inline src/main/java/application/fxobjects/cell/graph/CollectionCell.java:26]: nucleotides parameter misses JDoc

File: src/main/java/core/GraphReducer.java [hunk 4]
```diff
@@ -165,6 +166,7 @@ public static Boolean collapseNodeSequence(HashMap<Integer, Node> nodeMap, Node
         parent.setType(CellType.COLLECTION);
         parent.setSequence("");
         parent.setCollapseLevel(totalCollapseLevel);
+        parent.setNucleotides(parent.getNucleotides()+child.getNucleotides());
```
Comment (ghost) [inline src/main/java/core/GraphReducer.java]: "+" will cause a checkstyle violation

File: src/main/java/core/Node.java [hunk 5]
```diff
@@ -57,6 +57,11 @@
     private List<String> genomes;
 
     /**
+     * Amount of nucleotides in the node.
+     */
+    private int nucleotides;
```
Comment (ghost) [inline src/main/java/core/Node.java:62]: This redundant as you can just access seq.length()

Even if you have a good reason to preserve it, the name 'nucleotides' does not make clear that it contains the number of nucleotides. In that case something like numNucleotides has to be used.

File: src/main/java/core/graph/Graph.java [hunk 6]
```diff
@@ -191,7 +191,7 @@ public Model generateModel(Object ref, int depth, Model toret) {
             // Only draw when the intersection > 0 (Node contains genome that we
             // want to draw.
             if (intersection(root.getGenomes(), currentGenomes) > 0) {
-                toret.addCell(root.getId(), root.getSequence(), CellType.RECTANGLE);
+                toret.addCell(root.getId(), root.getSequence(),root.getNucleotides(), CellType.RECTANGLE);
```
Comment (ghost) [inline src/main/java/core/graph/Graph.java]: This has to be replaced by root.getSequence().length()

File: src/main/java/core/graph/Graph.java [hunk 7]
```diff
@@ -248,19 +248,18 @@ public void addCell(HashMap<Integer, Node> nodeMap, Model toret, int j,
...
 
         if (type == CellType.RECTANGLE) {
-            toret.addCell(to.getId(), to.getSequence(),
+            toret.addCell(to.getId(), to.getSequence(), to.getNucleotides(),
                     CellType.RECTANGLE);
-
         } else if (type == CellType.BUBBLE) {
             toret.addCell(to.getId(),
-                    Integer.toString(to.getCollapseLevel()),
+                    Integer.toString(to.getCollapseLevel()),to.getNucleotides(),
```
Comment (ghost) [inline src/main/java/core/graph/Graph.java]: Missing space --> will cause checkstyle violation

File: src/main/java/core/graph/PhylogeneticTree.java [hunk 8]
```diff
@@ -77,9 +77,9 @@ public void setup(Tree tree) {
 
         for (TreeNode node : tree.nodes) {
             if (node.isLeaf()) {
-                model.addCell(node.getKey(), node.getName(), CellType.TREELEAF);
+                model.addCell(node.getKey(), node.getName(), 0, CellType.TREELEAF);
             } else {
-                model.addCell(node.getKey(), node.getName(), CellType.TREEMIDDLE);
+                model.addCell(node.getKey(), node.getName(),0, CellType.TREEMIDDLE);
```
Comment (Warnars) [inline src/main/java/core/graph/PhylogeneticTree.java]: Missing space

**Comment by tieswesterborg** [comment]
No further remarks.

**Pull request merged**